lib/spandex_datadog/adapter.ex

```diff
@@ -106,9 +106,7 @@ defmodule SpandexDatadog.Adapter do
     end)
     |> String.to_integer()
   rescue
-    e ->
-      Logger.error("Failed to parse W3C priority, tracestate: #{inspect(tracestate)}, error: #{inspect(e)}")
-      @default_priority
+    _ -> @default_priority
```

Comment:
This will catch all errors, including future ones. It's better to handle the single one that you expect here and any future ones will still be caught.

Reply:
Maybe add a test for that:
```
%ArgumentError{message: "errors were found at the given arguments:

  * 1st argument: not a binary
"}
```

Reply:
It's a good approach to distinguish between two scenarios: when the priority parameter is missing versus when it fails to parse (for example, due to an invalid tracestate). Only parsing failures should generate error logs – and I changed it to work like this.